### PR TITLE
Bk/animation offscreen items improvement

### PR DESCRIPTION
**Pull request by bryankeller**
## Details

This improves animations that involve views animating in / out of view. The root issue is that offscreen views get hidden from the view hierarchy. In order to work around this, we do the following:

- Detect that we're animating in `setContent`
- Internally force layout immediately with an _extended bounds_ that gets "offscreen" views visible (even though they're still technically out of view)
- Let `setContent` finish running, updating internal state to contain the latest content values
- The feature code / caller's `calendarView.layoutIfNeeded()` is called, forcing a layout with _extended bounds_ with the new content values in an animation closure
- After this animated layout pass, we stop using _extended bounds_ during layout and return to normal behavior of hiding views that are not visible

| Before | After |
| ---- | ---- |
| Simulator Screen Recording - iPhone 14 Pro - 2023-08-23 at 18 10 56 | Simulator Screen Recording - iPhone 14 Pro - 2023-08-23 at 18 09 52 | 

## Related Issue

N/A

## Motivation and Context

Animation improvements

## How Has This Been Tested

Example app

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: Sources/Public/CalendarView.swift [hunk 1]
```diff
@@ -707,6 +668,87 @@ public final class CalendarView: UIView {
...
+    }
+
+    cachedAccessibilityElements = nil
+    if let element = focusedAccessibilityElement as? OffScreenCalendarItemAccessibilityElement {
+      UIAccessibility.post(
+        notification: .screenChanged,
+        argument: visibleViewsForVisibleItems[element.correspondingItem])
+    }
+  }
+
```
Comment (bryankeller) [inline Sources/Public/CalendarView.swift:751]: this is just a copy-paste from the original `layoutSubviews` implementation. No logic change in these lines.

File: Sources/Public/CalendarView.swift [hunk 2]
```diff
@@ -198,60 +198,10 @@ public final class CalendarView: UIView {
...
-        minimumScrollOffset: minimumScrollOffset,
-        maximumScrollOffset: maximumScrollOffset)
-    }
-
-    cachedAccessibilityElements = nil
-    if let element = focusedAccessibilityElement as? OffScreenCalendarItemAccessibilityElement {
-      UIAccessibility.post(
-        notification: .screenChanged,
-        argument: visibleViewsForVisibleItems[element.correspondingItem])
-    }
```
Comment (bryankeller) [inline Sources/Public/CalendarView.swift:254]: moved to https://github.com/airbnb/HorizonCalendar/pull/260/files#diff-733b4844a228b4fda6b27e4f88b42a6fc4b04eb589081da658741416a2152bd9R696-R751 with no logic change

**Review by brynbodayle — APPROVED** [review]

**Review by brynbodayle — APPROVED** [review]

File: Sources/Public/CalendarView.swift [hunk 3]
```diff
@@ -527,6 +486,8 @@ public final class CalendarView: UIView {
   private var visibleItemsDetails: VisibleItemsDetails?
   private var visibleViewsForVisibleItems = [VisibleItem: ItemView]()
 
+  private var isAnimatedUpdatePass = false
```
[the remark this answers is not in the document]
Comment (bryankeller) [reply Sources/Public/CalendarView.swift:489]: Updated!

[the remark this answers is not in the document]
Comment (brynbodayle) [reply Sources/Public/CalendarView.swift:489]: remove this buddy?

**Pull request merged**